### PR TITLE
fix: synchronize before gradient clipping in PyTorch

### PR DESCRIPTION
**Pull request by aaron276h**
## Description
Previously when not using mixed precision, we were potentially performing gradient clipping prior to gradient communication being completed.

## Test Plan
Going to run the full suite tests before merging.

### CODE REVIEW

**Comment by aaron276h** [comment]
All the tests passed: https://app.circleci.com/pipelines/github/determined-ai/determined/1911/workflows/faf45d76-00c2-4197-bbf7-4e8d4415a300
